<?xml version="1.0" ?>
<launch>


  <arg name="db_path" default="/home/adam/mongodb"/>
  <arg name="use_rviz" default="true"/>

  <!-- Robot arguments -->
  <arg name="robot_model" default="$(find rasberry_bringup)/config/robot_007.yaml"/>
  <arg name="model_extras" default="$(find rasberry_bringup)/urdf/robot_007_sensors.xacro"/>

  <!-- Gazebo arguments -->
  <arg name="world_name" default="riseholme" />
  <arg name="with_gui" default="true" />

  <!-- Map arguments -->
  <arg name="map" default="$(find rasberry_navigation)/maps/riseholme_sim.yaml" />
  <arg name="use_no_go_map" default="true" />
  <arg name="no_go_map" default="$(find rasberry_navigation)/maps/riseholme_sim_no_go.yaml" />
  <arg name="topo_map" default="riseholme_sim" />

  <!-- Move base arguments -->
  <arg name="use_carrot_planner" default="false"/>

  <!-- Topological navigation arguments -->
  <arg name="move_base_reconf_service" default="DWAPlannerROS" />



 
  <!-- Mongo database -->
  <include file="$(find mongodb_store)/launch/mongodb_store.launch">
    <arg name="db_path" value="$(arg db_path)" />
  </include>

  <!-- Robot and Gazebo bringup -->
  <include file="$(find rasberry_bringup)/launch/robot_bringup.launch">
    <arg name="robot_model" value="$(arg robot_model)" />
    <arg name="model_extras" value="$(arg model_extras)" />
    <arg name="simple_sim" value="true" />
    <arg name="world_name" value="$(arg world_name)" />
    <arg name="with_gui" value="$(arg with_gui)" />
  </include>

  <!-- Metric maps -->
  <include file="$(find rasberry_navigation)/launch/map_server.launch">
    <arg name="map" value="$(arg map)" />
    <arg name="use_no_go_map" value="$(arg use_no_go_map)" />
    <arg name="no_go_map" value="$(arg no_go_map)" />
  </include>

  <!-- Topological map -->
  <include file="$(find rasberry_navigation)/launch/topological_map_manager_central.launch">
    <arg name="map" value="$(arg topo_map)" />
  </include>

  <!-- amcl -->
  <include file="$(find rasberry_navigation)/launch/amcl.launch"/>

  <!-- Move base bringup -->
  <include file="$(find rasberry_move_base)/launch/move_base_dwa.launch">
    <arg name="use_carrot_planner" value="$(arg use_carrot_planner)"/>
  </include>

  <!-- Topological navigation bringup -->
  <include file="$(find rasberry_navigation)/launch/topological_navigation_robot.launch">
    <arg name="move_base_reconf_service" value="$(arg move_base_reconf_service)" />
  </include>

  <!-- Run rviz -->
  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find rasberry_bringup)/resources/topological_navigation.rviz" if="$(arg use_rviz)"/>


</launch>
